### PR TITLE
Fixes #82

### LINKED ISSUE #82: Support Turn LongTermCredentials

### PR DESCRIPTION
**Pull request by 1-alex98**

### CODE REVIEW

File: faf-java-server-app/src/main/java/com/faforever/server/ice/StandardIceServerProvider.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.time.Instant;
+import java.util.Base64;
+import java.util.stream.Collectors;
+
+@Service
+@Slf4j
+@Conditional(StandardIceServerProviderCondition.class)
+public class StandardIceServerProvider implements IceServersProvider {
```
Comment (Geosearchef) [inline faf-java-server-app/src/main/java/com/faforever/server/ice/StandardIceServerProvider.java]: What about LongTormCrendential(IceServer)Provider?

File: faf-java-server-app/src/main/java/com/faforever/server/ice/StandardIceServerProvider.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  }
+
+  @Override
+  public IceServerList getIceServerList(Player player) {
+    Ice ice = properties.getIce();
+    int ttl = ice.getTtl();
+
+    String userAuth = String.format("%s:%s", Instant.now().plusSeconds(ttl).getEpochSecond(), player.getLogin());
+
+    return new IceServerList(ttl, Instant.now(),
```
Comment (Geosearchef) [inline faf-java-server-app/src/main/java/com/faforever/server/ice/StandardIceServerProvider.java]: this and the userAuth/name should probably have the same time if someone tries to calculate the date of expiry using this and the ttl.

File: faf-java-server-app/src/main/java/com/faforever/server/ice/StandardIceServerProvider.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+  private String createToken(String userAuth, String secret) {
+    byte[] token;
+    try {
+      Mac mac = Mac.getInstance("HmacSHA1");
+      mac.init(new SecretKeySpec(Charset.forName("cp1252").encode(secret).array(), "HmacSHA1"));
+      token = mac.doFinal(Charset.forName("cp1252").encode(userAuth).array());
+
+    } catch (NoSuchAlgorithmException | InvalidKeyException e) {
+      log.error("Could not build secret key", e);
+      throw new RuntimeException(e);
```
Comment (Geosearchef) [inline faf-java-server-app/src/main/java/com/faforever/server/ice/StandardIceServerProvider.java]: Is this copied from my test code? I don't think we should just throw a runtime exception here, instead just don't return an ICE server for this url and log an error.

**Comment by Geosearchef** [comment]
Thx for implementing this!

File: faf-java-server-app/src/main/java/com/faforever/server/ice/StandardIceServerProvider.java (hunk 3 above)
Comment (1-alex98) [reply faf-java-server-app/src/main/java/com/faforever/server/ice/StandardIceServerProvider.java]: yes it is and you even did System.exit(-1) but i thought that is a little over dramatic

File: faf-java-server-app/src/main/java/com/faforever/server/ice/StandardIceServerProvider.java (hunk 1 above)
Comment (1-alex98) [reply faf-java-server-app/src/main/java/com/faforever/server/ice/StandardIceServerProvider.java]: What is that????

**Comment by 1-alex98** [comment]
I made you a new image with the bug fixed "ice-test"

File: faf-java-server-app/src/main/java/com/faforever/server/ice/StandardIceServerProvider.java (hunk 1 above)
Comment (Geosearchef) [reply faf-java-server-app/src/main/java/com/faforever/server/ice/StandardIceServerProvider.java]: A better name than standard ice server provider ;)

**Comment by 1-alex98** [comment]
Anymore discusions on this?

**Comment by micheljung** [comment]
No, all good AFAIK. Build fails, waiting for https://github.com/travis-ci/travis-ci/issues/10249 to be fixed

**Pull request merged**